fili-core/src/main/java/com/yahoo/bard/webservice/metadata/DataSourceMetadataService.java

```diff
@@ -87,15 +124,40 @@ public void update(PhysicalTable table, DataSourceMetadata metadata) {
     /**
      * Get all the segments associated with the given Set of physical tables.
      *
-     * @param physicalTables  A Set of physicalTables used by the DruidQuery
+     * @param physicalTableNames  A Set of physical TableNames used by the DruidQuery
      *
      * @return A set of segments associated with the given tables
      */
-    public Set<SortedMap<DateTime, Map<String, SegmentInfo>>> getTableSegments(Set<PhysicalTable> physicalTables) {
-        return physicalTables.stream()
-                .map(allSegments::get)
+    public Set<SortedMap<DateTime, Map<String, SegmentInfo>>> getTableSegments(Set<TableName> physicalTableNames) {
+        return physicalTableNames.stream()
+                .map(allSegmentsByTime::get)
                 .filter(Objects::nonNull)
                 .map(AtomicReference::get)
                 .collect(Collectors.toSet());
     }
+
+    /**
+     * Get a set of intervals available for each column in the table.
+     *
+     * @param physicalTableName the table to get the column and availability for
+     *
+     * @return a map of column name to a set of avialable intervals
+     */
+    public Map<String, Set<Interval>> getAvailableIntervalsByTable(TableName physicalTableName) {
+        try {
+            return allSegmentsByColumn.get(physicalTableName).get();
+
+        } catch (NullPointerException e) {
```

Comment:
This will get rid of this Codacy complaint and maybe be a little more explicit about which thing is plausibly uninitialized.
(Looking at the code, we could access a table's segments before the DSMService has loaded it, and we'd have the table missing from the map.  But if it has ever been loaded nulls should not happen)

```
   public Map<String, Set<Interval>> getAvailableIntervalsByTable(TableName physicalTableName) {
        if (allSegmentsByColumn.containsKey(physicalTableName)) {
            return allSegmentsByColumn.get(physicalTableName).get();
        } else {
            LOG.error(
                    "Trying to access {} physical table datasource that is not available in metadata service",
                    physicalTableName.asName()
            );
            throw new IllegalStateException(
                    String.format(
                            "Trying to access %s physical table datasource that is not available in metadata service",
                            physicalTableName.asName()
                    )
            );
        }
    }
```